### ISSUE TITLE
Redis connection timeouts with v2.4.0

### ISSUE THREAD

**Issue by rafaelmaeuer** [DataDog/dd-trace-js#1937]
**Describe the bug**

After upgrading to v2.4.0 docker-pods are crashing with redis `Connection timeouts`:
```
Error: Connection timeout
    at Socket.<anonymous> (/web/app/node_modules/@node-redis/client/dist/lib/client/socket.js:169:124)
    at /web/app/node_modules/dd-trace/packages/dd-trace/src/scope.js:65:19
    at Scope.activate (/web/app/node_modules/dd-trace/packages/dd-trace/src/scope.js:29:14)
    at Socket.bound (/web/app/node_modules/dd-trace/packages/dd-trace/src/scope.js:64:20)
    at Object.onceWrapper (node:events:645:28)
    at Socket.emit (node:events:538:35)
    at Socket.emit (node:domain:475:12)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
```

I reverted back to v2.3.1 but the problem remained. Only after reverting back peer-dependency `datadog/native-appsec` from `0.8.3` to `0.8.1` too, pods are behaving normal again. So the problem might lie there...

**Environment**

* **Operation system:** Docker-Alpine
* **Node version:** 16.14.2
* **Tracer version:** 2.4.0
* **Agent version:** ???

**Comment by mhassan1** [comment]
@rafaelmaeuer Is this a duplicate of https://github.com/DataDog/dd-trace-js/issues/1916?

**Comment by rochdev** [comment]
We've rewritten the `net` plugin which was involved in the above stack trace. Can you try with 2.4.1?

**Comment by frytg** [comment]
Version 2.4.1 seems to be working again (on the system from @rafaelmaeuer's problem above)!
